src/charm.py

```diff
@@ -34,152 +21,106 @@
     ResourceRequirements,
     adjust_resource_requirements,
 )
-from charms.observability_libs.v1.kubernetes_service_patch import KubernetesServicePatch
```

Comment:
question(nit): since this is removed, should we also remove the lib from the `libs` folder?